File: src/1306-Jump-Game-III/1306.go
func canReach(arr []int, start int) bool {
    q := []int{start}
    vis := make([]bool, len(arr))
    vis[start] = true
    
    for len(q) > 0 {
        cur := q[0]
        q = q[1:]
        if arr[cur] == 0 {
            return true
        }
        
        vis[cur] = true
        a, b := cur + arr[cur], cur - arr[cur]
        if a >= 0 && a < len(arr) && !vis[a] {
            q = append(q, a)
        }
        if b >= 0 && b < len(arr) && !vis[b] {
            q = append(q, b)
        }
    }
    return false
}

File: src/1306-Jump-Game-III/1306.js
var canReach = function(arr, start) {
    let q = [start], vis = new Array(arr.length);
    vis.fill(0);
    vis[start] = 1;
    
    while (q.length) {
        let cur = q.shift();
        if (arr[cur] == 0) return true;
        
        vis[cur] = 1;
        let a = cur + arr[cur], b = cur - arr[cur];
        if (a >= 0 && a < arr.length && !vis[a]) q.push(a);
        if (b >= 0 && b < arr.length && !vis[b]) q.push(b);
    }
    return false;
};